src/ripple/rpc/impl/NFTokenID.cpp

```diff
@@ -0,0 +1,200 @@
+    // NFT IDs and compare them to find what changed.
+    std::vector<uint256> prevIDs;
+    std::vector<uint256> finalIDs;
+
+    for (STObject const& node : transactionMeta.getNodes())
+    {
+        if (node.getFieldU16(sfLedgerEntryType) != ltNFTOKEN_PAGE)
+            continue;
+
+        SField const& fName = node.getFName();
+        if (fName == sfCreatedNode)
+        {
+            STArray const& toAddPrevNFTs = node.peekAtField(sfNewFields)
+                                               .downcast<STObject>()
+                                               .getFieldArray(sfNFTokens);
+            std::transform(
+                toAddPrevNFTs.begin(),
+                toAddPrevNFTs.end(),
+                std::back_inserter(finalIDs),
+                [](STObject const& nft) {
+                    return nft.getFieldH256(sfNFTokenID);
+                });
+        }
+        else if (fName == sfModifiedNode)
+        {
+            // When a mint results in splitting an existing page,
+            // it results in a created page and a modified node. Sometimes,
+            // the created node needs to be linked to a third page, resulting
+            // in modifying that third page's PreviousPageMin or NextPageMin
+            // field changing, but no NFTs within that page changing. In this
+            // case, there will be no previous NFTs and we need to skip.
+            // However, there will always be NFTs listed in the final fields,
+            // as rippled outputs all fields in final fields even if they were
+            // not changed.
+            STObject const& previousFields =
+                node.peekAtField(sfPreviousFields).downcast<STObject>();
+            if (!previousFields.isFieldPresent(sfNFTokens))
+                continue;
+
+            STArray const& toAddPrevNFTs =
+                previousFields.getFieldArray(sfNFTokens);
+            std::transform(
+                toAddPrevNFTs.begin(),
+                toAddPrevNFTs.end(),
+                std::back_inserter(prevIDs),
+                [](STObject const& nft) {
+                    return nft.getFieldH256(sfNFTokenID);
+                });
+
+            STArray const& toAddFinalNFTs = node.peekAtField(sfFinalFields)
+                                                .downcast<STObject>()
+                                                .getFieldArray(sfNFTokens);
+            std::transform(
+                toAddFinalNFTs.begin(),
+                toAddFinalNFTs.end(),
+                std::back_inserter(finalIDs),
+                [](STObject const& nft) {
+                    return nft.getFieldH256(sfNFTokenID);
+                });
+        }
+    }
+
+    // We expect NFTs to be added one at a time.  So finalIDs should be one
+    // longer than prevIDs.  If that's not the case something is messed up.
+    if (finalIDs.size() != prevIDs.size() + 1)
+        return std::nullopt;
```

Comment:
@ledhed2222, that's a fair question.  As a sanity check I just tried
```
grep -r -o std\:\:nullopt src/ripple | wc -l
```
That gave me over 250 hits of uses of `std::nullopt` in the code base.

Personally, I usually prefer things like `std::nullopt` over `{}` because when I read the source code (as text) I know what they are.  `{}` can be anything with a default constructor.  But I also know not everyone agrees with me.

Reply:
i think ill go with `nullopt`. rippled does use it fairly common